vertx-web-openapi-router/src/main/java/io/vertx/ext/web/openapi/router/impl/AuthenticationHandlers.java

```diff
@@ -0,0 +1,170 @@
+package io.vertx.ext.web.openapi.router.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.AuthenticationHandler;
+import io.vertx.ext.web.handler.ChainAuthHandler;
+import io.vertx.ext.web.handler.OAuth2AuthHandler;
+import io.vertx.ext.web.handler.SimpleAuthenticationHandler;
+import io.vertx.ext.web.handler.impl.ScopedAuthentication;
+import io.vertx.openapi.contract.OpenAPIContract;
+import io.vertx.openapi.contract.Operation;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Francesco Guardiani @slinkydeveloper
+ */
+class AuthenticationHandlers {
+
+  private static final JsonObject EMPTY_JSON = new JsonObject(Collections.emptyMap());
+
+  private static final AuthenticationHandler SUCCESS_HANDLER =
+    SimpleAuthenticationHandler
+      .create()
+      .authenticate(ctx -> Future.succeededFuture(User.create(EMPTY_JSON)));
+
+  private final Map<String, List<AuthenticationHandler>> securityHandlers;
+  private final Map<String, OAuth2AuthHandler> callbackHandlers;
+
+  AuthenticationHandlers() {
+    this.securityHandlers = new HashMap<>();
+    this.callbackHandlers = new HashMap<>();
+  }
+
+  protected void addRequirement(String name, AuthenticationHandler handler, String callback) {
+    securityHandlers
+      .computeIfAbsent(name, k -> new ArrayList<>())
+      .add(handler);
+
+    if (callback != null) {
+      // TODO: check if callback is already present
+      callbackHandlers.put(callback, (OAuth2AuthHandler) handler);
+    }
+  }
+
+  /**
+   * The input array is an OR of different AND security requirements
+   */
+  protected void solve(OpenAPIContract contract, Operation operation, Route route, boolean failOnNotFound) {
```

Comment:
This should be solved already in OpenAPIContract

Reply:
As discussed, this can't be really solved in OpenAPIContract. But we could already resolve the effective SecurityRequirement for an operation [1]. Then we don't have to pass the OpenAPIContract.

[1] https://github.com/eclipse-vertx/vertx-openapi/pull/19#discussion_r1140182963